### PR TITLE
Start wgsl-out

### PR DESCRIPTION
**Pull request by Gordon-F**
Start implementation of `wgsl-out`. Just a copy-paste `hlsl-out` 😄

### CODE REVIEW

**Review by kvark — COMMENTED** [review]
Just one question, looks great overall!
I imagine, to validate it, we are going to feeding it back into wgsl-in? :)

**Comment by kvark** [comment]
Let's follow-up with `fmt::Error`, `fmt::Write`, and friends

**Review by kvark — APPROVED** [review]

File: src/back/wgsl/writer.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
+use super::Error;
+use crate::FastHashMap;
+use crate::{
+    back::wgsl::keywords::RESERVED, Function, Module, ShaderStage, StructLevel, TypeInner,
+};
+use crate::{
+    proc::{NameKey, Namer},
+    StructMember,
+};
+use std::io::Write;
```
[the remark this answers is not in the document]
Comment (Gordon-F) [reply src/back/wgsl/writer.rs:10]: Should we do the same for `glsl-out`?

[the remark this answers is not in the document]
Comment (kvark) [reply src/back/wgsl/writer.rs:10]: Yes, for everything. fmt::Write is strictly more general, and fmt::Error is smaller in size

**Pull request merged**